### PR TITLE
Upgrade to version 7.3.0

### PR DESCRIPTION
**Pull request by grossmannmartin**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Upgrade demoshop to Shopsys Framework 7.3.0
|New feature| Yes 
|Fixes issues| ...
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

**Review by vitek-rostislav — COMMENTED** [review]
Hi, all the current changes seem ok to me, I have just extracted one change into a separate commit.

However, I am not able to build the application locally, it fails on error when running `product-search-create-structure`:
```
ERROR     [app] Elasticsearch\Common\Exceptions\BadRequest400Exception: {"error":{"root_cause":[{"type":"mapper_parsing_exception","reason":"no handler for type [icu_collation_keyword] declared on field [keyword]"}],"type":"mapper_parsing_exception","reason":"Failed to parse mapping [_doc]: no handler for type [icu_collation_keyword] declared on field [keyword]","caused_by":{"type":"mapper_parsing_exception","reason":"no handler for type [icu_collation_keyword] declared on field [keyword]"}},"status":400} (uncaught exception) at /var/www/html/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php line 620 while running console command `shopsys:product-search:create-structure` ["exception" => Elasticsearch\Common\Exceptions\BadRequest400Exception { …}]
```

It is related to `icu_collation_keyword`, it works on CI, and I have a strong feeling it is caused by cached Docker images or something like that, so I am running a clean installation at the moment, I will let you know later whether it helped.

**Comment by vitek-rostislav** [comment]
Hi, on my machine, the error is there even after the clean installation. Could you please confirm that it works on your local? Do you have any clue what am I missing? :confused:

**Comment by grossmannmartin** [comment]
Hi, it really works on my machine, but I installed it locally. Can you take a look at docker container running Elasticsearch? If it’s created properly from dockerfile? I presume docker-compose file is valid.

**Comment by vitek-rostislav** [comment]
Hi, everything works now, it seems like I have been using a wrong version of `docker-compose.yml`, my bad :man_facepalming:

**Review by boris-brtan — APPROVED** [review]

File: docker/conf/docker-compose.demo.yml.dist [hunk 1]
```diff
@@ -49,4 +49,4 @@ volumes:
 networks:
     shared-elastic:
         external:
-            name: ${ELASTICSEARCH_NETWORK:-shared-demo-elasticsearch-network}
+            name: ${ELASTICSEARCH_NETWORK:-shared-demo-elasticsearch-icu-instance}
```
Comment (boris-brtan) [inline docker/conf/docker-compose.demo.yml.dist]:
```suggestion
            name: ${ELASTICSEARCH_NETWORK:-shared-demo-elasticsearch-icu-network}
```

**Review by boris-brtan — CHANGES_REQUESTED** [review]
THX,
just one little thing

**Review by boris-brtan — APPROVED** [review]

**Pull request merged**